### ISSUE TITLE
[Bug] Broker AcquireTokenInteractive() doesn't cache when it fallbacks

### ISSUE THREAD

**Issue by LennoxP90** [AzureAD/microsoft-authentication-library-for-dotnet#4209]
**Logs and network traces**
{MSAL.Xamarin.Android.4.54.1.0.MsalUiRequiredException: 
	ErrorCode: failed_to_acquire_token_silently_from_broker
Microsoft.Identity.Client.MsalUiRequiredException: Broker could not satisfy the silent request.
  at Microsoft.Identity.Client.Internal.Requests.Silent.SilentRequest.ExecuteAsync (System.Threading.CancellationToken cancellationToken) [0x002d7] in <f5e0daf1cb4c43c889fca0f3b0814e85>:0 
  at Microsoft.Identity.Client.Internal.Requests.RequestBase.RunAsync (System.Threading.CancellationToken cancellationToken) [0x001c5] in <f5e0daf1cb4c43c889fca0f3b0814e85>:0 
  at Microsoft.Identity.Client.ApiConfig.Executors.ClientApplicationBaseExecutor.ExecuteAsync (Microsoft.Identity.Client.ApiConfig.Parameters.AcquireTokenCommonParameters commonParameters, Microsoft.Identity.Client.ApiConfig.Parameters.AcquireTokenSilentParameters silentParameters, System.Threading.CancellationToken cancellationToken) [0x00144] in <f5e0daf1cb4c43c889fca0f3b0814e85>:0 
  at WiFiMessenger.Services.Communication.OIDCToken.StartRefreshingTokensAsync () [0x0010a] in F:\Code\Work\Wi-Fi Messenger\Client\Client\WiFiMessenger\Services\Communication\OIDCToken.cs:189 
	StatusCode: 0 
	ResponseBody:  
	Headers: }

**Which version of MSAL.NET are you using?**
4.54.1

**Platform**
Xamarin.Android

**What authentication flow has the issue?**
* Desktop / Mobile
    * [x] Interactive (broker)
    * [x] Interactive (embedded or system browser)
    * [ ] Integrated Windows Authentication
    * [ ] Username / Password
    * [ ] Device code (browserless)
* Web app 
    * [ ] Authorization code
    * [ ] On-Behalf-Of
* Daemon app 
    * [ ] Client credentials (service-to-service calls)

**Is this a new or existing app?**
new

**Actual behavior**
When I get a token using AcquireTokenInteractive with my PublicClientApplication built WithBroker(), and the device does not have a broker installed, so it should fall back to webview. When I get that token I try to refresh it periodically using AcquireTokenSilent and the account hint that was in the token. The acquire fails and gives above exception. When I try the same with the WithBroker() commented out, so not using broker flow. Everything works as expected.

**Expected behavior**
use the cached token that was gathered seconds ago using the interactive login

**Possible solution**
If i call 
```csharp
.WithBroker( PublicClientApplicationBuilder.Create( applicationId ).IsBrokerAvailable());
```
I understand this is deprecated but it works for me in the meantime until there is an official fix

**Comment by bgavrilMS** [comment]
Let me rephrase to see if I understand the repro steps. 

`WithBroker = true`

1. AcquireTokenInteractive but broker is not installed, so a webview is used (internally this makes MSAL cache a token)
2. AcquireTokenSilent works as long as broker is not installed
3. Later, the broker gets installed
4. AcquireTokenSilent stops working with the error message above

**Comment by LennoxP90** [comment]
no, if the broker never is installed but I attempt to use the broker flow the falled back webview does not cache the account. If the broker is indeed installed it works fine.

**Comment by LennoxP90** [comment]
so your repro would be as follows:

`WithBroker = true`

1. AcquireTokenInteractive but broker is not installed, so a webview is used (internally this **should** make MSAL cache a token)
2. AcquireTokenSilent fails with the above error message
3. Later, the broker gets installed
4. AcquireTokenSilent starts working

**Comment by michel-douma** [comment]
We're having the exact same problem. Also using version 4.54.1. The workaround which is provided by @LennoxP90 works in the meanwhile although this is indeed deprecated. The deprecation message `This method is obsolete. Applications should rely on the library automatically falling back to a browser if the broker is not available.` is kind of misleading since can't rely on the library to autimatically fall back at this time. Hope this will be fixed soon.

**Comment by bgavrilMS** [comment]
Ack, we will aim to get this fixed. 

Please use the IsBrokerAvailable, it should work fine, but we just didn't want app developers to use it, as MSAL should fallback to browser on its own.

**Comment by bgavrilMS** [comment]
Aiming to create a board that has mobile / desktop scenarios, separate from web site / web api scenarios.

**Comment by bgavrilMS** [comment]
I think this is higher importance than other bugs, consider this being a P1.

**Comment by LennoxP90** [comment]
has any headway been put on this?

**Comment by bgavrilMS** [comment]
@ashok672 - let's discuss this on during a standup. I think it needs to be prioritized.
